doc/usecases/elektra_web/UC_key_validation.md

```diff
@@ -0,0 +1,25 @@
+# Use Case: Key validation
+
+## Summary
+
+Title: Key validation
+Scope: Configuration
+Level: User Goal
+Actors: User (usually a sysadmin)
+Brief: User gets feedback about a key's contents.
+
+## Scenarios
+
+Precondition: Modifying keys
+Main success scenario: User edits a key's contents in a field that is
+  specifically formatted according to its metadata (e.g. number/date fields vs
+  string fields with regex validation). If the entered data passes the
+  validation, it is sent to the backend and saved to the key database.
+Alternative scenario: User enters data that fails the validation, the input
+  field is marked as invalid (e.g. with a red color and some information text)
+  and the data is NOT sent to the backend.
+Error scenario: Technical problems while persisting to the key database. The
```

Comment:
Maybe I'm misunderstanding something, but while undo/redo is definitely its own use case (it's a separate feature that is triggered with certain buttons), validation is a feature that is integrated in all use cases that include text fields.

That's also what makes it hard to have a definitive order of preconditions - because validation happens until the field is valid, in which case it is stored on the server.

If I was gonna make it as a separate use case, I would have to copy & paste the same validation use case for various text fields, with little differences. So the best way to go about this would be explaining validation in the modifying keys use case, but also in the adding keys use case, and others.